File: src-tauri/src/domain/publish.rs
// ─── Publish 领域类型 ───

use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct PlatformConfig {
    pub id: String,
    pub platform: String,
    pub label: String,
    pub app_id: String,
    pub app_secret: String,
    pub access_token: Option<String>,
    pub token_expires_at: Option<u64>,
    pub enabled: bool,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct PublishRecord {
    pub id: String,
    pub article_id: String,
    pub platform: String,
    pub platform_article_id: Option<String>,
    pub status: String,
    pub error_message: Option<String>,
    pub published_at: u64,
    pub platform_url: Option<String>,
}

File: src/domain/Publish.ts
// domain/Publish.ts — 发布相关纯类型定义

/** 平台配置 */
export interface PlatformConfig {
  id: string;
  platform: string;
  label: string;
  appId: string;
  appSecret: string;
  accessToken?: string;
  tokenExpiresAt?: number;
  enabled: boolean;
}

/** 发布选项 */
export interface PublishOptions {
  title?: string;
  coverImage?: string;
  summary?: string;
  declareOriginal: boolean;
  allowReprint: boolean;
  chargeable: boolean;
  author?: string;
  contentSourceUrl?: string;
  picCrop2351?: string;
  picCrop11?: string;
  productKey?: string;
}

/** 发布结果 */
export interface PublishResult {
  success: boolean;
  platformArticleId?: string;
  platformUrl?: string;
  errorMessage?: string;
  isDraft: boolean;
}
